TestExpireEDUs is racy when using SQLite

<!--
All bug reports must provide the following background information
Text between <!-- and --​> marks will be invisible in the report.
-->

### Background information
<!-- Please include versions of all software when known e.g database versions, docker versions, client versions -->
- **Dendrite version or git SHA**: ad4ac2c016d2916c51695fd11d3a7b52e38d0ca2 and 0.9.3
- **Monolith or Polylith?**: n/a
- **SQLite3 or Postgres?**: SQlite3
- **Running in Docker?**: no
- **`go version`**: 1.18.5 and 1.18.2
- **Client used (if applicable)**: n/a


### Description

 - **What** is the problem: the `TestExpireEDUs` test is racy when using SQLite
 - **Who** is affected: developers, package maintainers
 - **How** is this bug manifesting: running the tests
 - **When** did this first appear: presumably 240ae257deb74b7be8a17500b77d5e1bca56e8f5

<!--
Examples of good descriptions:
- What: "I cannot log in, getting HTTP 500 responses"
- Who: "Clients on my server"
- How: "Errors in the logs saying 500 internal server error"
- When: "After upgrading to 0.3.0"

- What: "Dendrite ran out of memory"
- Who: "Server admin"
- How: "Lots of logs about device change updates"
- When: "After my server joined Matrix HQ"


Examples of bad descriptions:
- What: "Can't send messages"  - This is bad because it isn't specfic enough. Which endpoint isn't working and what is the response code? Does the message send but encryption fail?
- Who: "Me" - Who are you? Running the server or a user on a Dendrite server?
- How: "Can't send messages" - Same as "What".
- When: "1 day ago" - It's impossible to know what changed 1 day ago without further input.
-->

### Steps to reproduce
<!-- Please try reproducing this bug before submitting it. Issues which cannot be reproduced risk being closed. -->

run `go test -run TestExpireEDUs/sqlite -count=3 ./federationapi/storage`:
```
$ go test -run TestExpireEDUs/sqlite -count=3 ./federationapi/storage
time="2022-08-18T03:45:20.848566000Z" level=info msg="Dendrite version 0.9.3"
time="2022-08-18T03:45:20.937088000Z" level=info msg="Dendrite version 0.9.3"
time="2022-08-18T03:45:20.937108000Z" level=info msg="Dendrite version 0.9.3"
--- FAIL: TestExpireEDUs (0.00s)
    --- FAIL: TestExpireEDUs/sqlite (0.07s)
        storage_test.go:64:
            	Error Trace:	storage_test.go:64
            	            				db.go:181
            	Error:      	Not equal:
            	            	expected: 1
            	            	actual  : 2
            	Test:       	TestExpireEDUs/sqlite
time="2022-08-18T03:45:21.007753000Z" level=info msg="Dendrite version 0.9.3"
time="2022-08-18T03:45:21.007773000Z" level=info msg="Dendrite version 0.9.3"
time="2022-08-18T03:45:21.007781000Z" level=info msg="Dendrite version 0.9.3"
FAIL
FAIL	github.com/matrix-org/dendrite/federationapi/storage	0.412s
FAIL
$ go test -run TestExpireEDUs/sqlite -count=3 ./federationapi/storage
ok  	github.com/matrix-org/dendrite/federationapi/storage	0.404s
$ go test -run TestExpireEDUs/sqlite -count=3 ./federationapi/storage
time="2022-08-18T03:45:35.846323000Z" level=info msg="Dendrite version 0.9.3"
time="2022-08-18T03:45:35.935419000Z" level=info msg="Dendrite version 0.9.3"
time="2022-08-18T03:45:35.935446000Z" level=info msg="Dendrite version 0.9.3"
--- FAIL: TestExpireEDUs (0.00s)
    --- FAIL: TestExpireEDUs/sqlite (0.07s)
        storage_test.go:64:
            	Error Trace:	storage_test.go:64
            	            				db.go:181
            	Error:      	Not equal:
            	            	expected: 1
            	            	actual  : 2
            	Test:       	TestExpireEDUs/sqlite
time="2022-08-18T03:45:36.005100000Z" level=info msg="Dendrite version 0.9.3"
time="2022-08-18T03:45:36.005118000Z" level=info msg="Dendrite version 0.9.3"
time="2022-08-18T03:45:36.005127000Z" level=info msg="Dendrite version 0.9.3"
FAIL
FAIL	github.com/matrix-org/dendrite/federationapi/storage	0.409s
FAIL
```

I've notably not been able to reproduce this when running the test with Postgres.

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from server logs, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.

Alternatively, please send logs to @kegan:matrix.org or @neilalexander:matrix.org
with a link to the respective Github issue, thanks!
-->

--- a/federationapi/storage/storage_test.go
+++ b/federationapi/storage/storage_test.go
@@ -31,7 +31,7 @@ func mustCreateFederationDatabase(t *testing.T, dbType test.DBType) (storage.Dat
 
 func TestExpireEDUs(t *testing.T) {
 	var expireEDUTypes = map[string]time.Duration{
-		gomatrixserverlib.MReceipt: time.Millisecond,
+		gomatrixserverlib.MReceipt: 0,
 	}
 
 	ctx := context.Background()

